### PR TITLE
Add RHEL 9.4 Variant

### PR DESCRIPTION
**Pull request by marmijo**
Add the following files to support a RHEL 9.4 variant.

- `manifest-rhel-9.4.yaml`
- `image-rhel-9.4.yaml -> image-rhel-coreos-9.yaml`
- `extensions-rhel-9.4.yaml -> extensions-rhel-coreos-9.yaml`

See: https://issues.redhat.com/browse/COS-2476

### CODE REVIEW

**Comment by travier** [comment]
- Missing an extensions manifest.
- If the image.yaml is the same one as the other one then you can make it a symlink.

File: manifest-rhel-9.4.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+     ln -s /usr/lib/issue /etc/issue.net
+
+     # Let's have a non-boring motd, just like CL (although theirs is more subdued
+     # nowadays compared to early versions with ASCII art).  One thing we do here
+     # is add --- as a "separator"; the idea is that any "dynamic" information should
+     # be below that.
+     # See: https://projects.engineering.redhat.com/browse/COREOS-1029
+     . /etc/os-release
+     cat > /etc/motd <<EOF
+     CentOS Stream CoreOS $VERSION
```
Comment (travier) [inline manifest-rhel-9.4.yaml:100]:
```suggestion
     RHEL CoreOS $VERSION
```

Let's fake the name.

File: manifest-rhel-9.4.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+       Part of OKD ${OPENSHIFT_VERSION}, SCOS is a Kubernetes native operating system
```
Comment (travier) [inline manifest-rhel-9.4.yaml:101]:
```suggestion
       Part of OpenShift ${OPENSHIFT_VERSION}, RHCOS is a Kubernetes native operating system
```

File: manifest-rhel-9.4.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+     set -xeo pipefail
+
+     # Tweak /usr/lib/os-release
+     grep -v "OSTREE_VERSION" /etc/os-release > /usr/lib/os-release.stream
+     OCP_RELEASE="4.15"
+     (
+     . /etc/os-release
+     cat > /usr/lib/os-release <<EOF
+     NAME="${NAME} CoreOS"
+     ID="scos"
```
Comment (travier) [inline manifest-rhel-9.4.yaml:46]:
```suggestion
     ID="rhcos"
```

**Comment by marmijo** [comment]
Thanks! I forced pushed an update based on the review.

File: manifest-rhel-9.4.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  - rhel-9.2-fast-datapath
+  # Include RHCOS 9 repo for oc, hyperkube
+  - rhel-9.2-server-ose-4.15
+
+# We include hours/minutes to avoid version number reuse
+automatic-version-prefix: "415.94.<date:%Y%m%d%H%M>"
+# This ensures we're semver-compatible which OpenShift wants
+automatic-version-suffix: "-"
+# Keep this is sync with the version in postprocess
+mutate-os-release: "4.15"
```
Comment (marmijo) [inline manifest-rhel-9.4.yaml:32]: Should I also change this to target 4.14 instead of 4.15?

**Comment by travier** [comment]
This branch is 4.15 so let's do that.

**Review by travier — APPROVED** [review]
/lgtm

**Pull request merged**